src/compiler/emitter.ts

```diff
@@ -300,14 +300,29 @@ namespace ts {
                             continue;
                         }
 
-                        writeLines(helper.text);
+                        writeHelper(writeLines, helper.text, makeUniqueName);
                         helpersEmitted = true;
                     }
                 }
             }
 
             return helpersEmitted;
         }
+
+        function writeHelper(writer: (text: string) => void, helper: EmitHelper["text"], makeUniqueName: UniqueNameHandler) {

```

Comment:
There's a lot of indirection going on here, and `writeHelper` only seems to be used once. It seems like it would be better to just keep this inline in `emitHelpers` rather than add more complexity through even more callbacks and indirection.